### PR TITLE
PixelPaint: Shift-key for squares and circles

### PR DESCRIPTION
**Pull request by ZyperPL**
PR adds method for changing point's one coordinate (X or Y) to make X and Y difference with other point's X and Y equal to the maximum difference. In short: it makes abs(X2 - X1) == abs(Y2 - Y1). 

Method is used in Rectangle Tool for making squares and Ellipse Tool for making circles

### CODE REVIEW

File: Userland/Applications/PixelPaint/EllipseTool.cpp [hunk 1]
```diff
@@ -78,6 +78,10 @@ void EllipseTool::on_mousemove(Layer*, MouseEvent& event)
         return;
 
     m_ellipse_end_position = event.layer_event().position();
+
+    if (event.layer_event().shift())
+        m_ellipse_end_position = m_ellipse_start_position.end_point_for_square_aspect_ratio(event.layer_event().position());
```
Comment (linusg) [inline Userland/Applications/PixelPaint/EllipseTool.cpp:83]: Seems odd to write the value twice in the shift case, how about:

```suggestion
    if (event.layer_event().shift())
        m_ellipse_end_position = m_ellipse_start_position.end_point_for_square_aspect_ratio(event.layer_event().position());
    else
        m_ellipse_end_position = event.layer_event().position();
```

**Review by linusg — CHANGES_REQUESTED** [review]
Please reword the commit messages to use imperative mood.

**Review by linusg — APPROVED** [review]

**Comment by mustafaquraish** [comment]
If you're adding a helper to get the other end point for an aspect ratio anyway, maybe it would make sense to have the function take in an arbitrary aspect ratio and compute the endpoint for that, instead of just hard-coding it to be only for squares? I'm imagining it would be useful if we want to restrict the drawing of shapes to some configurable aspect ratio as well, in the future.

**Comment by TobyAsE** [comment]
@mustafaquraish this is a good idea, however I think let's put it into a future pr ;)

**Review by bgianfo — APPROVED** [review]
LGTM! 🔲 ⚪ 🖌️

**Pull request merged**